packages/runtime/container-runtime/src/pendingStateManager.ts

```diff
@@ -187,27 +189,30 @@ export class PendingStateManager implements IDisposable {
 	public readonly dispose = () => this.disposeOnce.value;
 
 	/**
-	 * Called when a message is submitted locally. Adds the message and the associated details to the pending state
-	 * queue.
-	 * @param type - The container message type.
-	 * @param content - The message content.
-	 * @param localOpMetadata - The local metadata associated with the message.
+	 * The given batch has been flushed, and needs to be tracked locally until the corresponding
+	 * acks are processed, to ensure it is successfully sent.
+	 * @param batch - The batch that was flushed
+	 * @param clientSequenceNumber - The CSN of the first message in the batch,
+	 * or undefined if the batch was not yet sent (e.g. by the time we flushed we lost the connection)
 	 */
-	public onSubmitMessage(
```

Comment:
Maybe this is me overthinking things, (non-blocking) why don't we just keep `onSubmitMessage` and call `onSubmitMessage` during `onFlushBatch` or even convert `onSubmitMessage` to `private`?

Reply:
Direction is to have PSM deal in while batches not individual messages